Few micro changes and optimizations from various authors

voituk opened the pull request:
Here is few changes in this batch:
1. Fixed bug in hash:changed event implementation
2. Cookie code optimizations from Victor Homaykov
3. Documentation micro-fix
4. Element.centerInParent implementation from some guy from Mexico.

Any comments? Objections?

Commit:
Changed Cookie#get

RumataMx committed:
Element.centerInParent could be adapted to center the element in another element, given that the container is relatively positioned and has defined dimentions, although this would mean that the element to be centered within another should be moved in the context of the mark up. Ha ve to do that later and refine the method.

voituk committed:
Updated array documentation

voituk committed:
Merged optimized version of Cookie.get

voituk committed:
Merge branch 'master' of github.com:voituk/protolicious

voituk committed:
Fixed hash_changed typos prevented the snippet to work.

voituk committed:
Fixed: this is unknown in this context

kangax commented:
Thanks for tackling these :) Few comments:
- What was the bug in hash:changed? Is it trailing `().call(null)` part?
- What is cookie code optimized for? Size? Does it also fix something? find + startsWith could make it slow (for long strings or those consisting of many values)
- Element#centerInParent sets style "manually"; could use `Element.setStyle`. I don't care much which one it is; depends on whether size matters more than speed or vice versa (so should be modified accordingly)

voituk committed:
Replaced direct style property usage with Element.setStyle

voituk commented:
Oh, it was so much time ago, so i don't remember all details :)
Anyway will try to describe you:
- Fixed hash_changed typos prevented the snippet to work. 
  https://github.com/voituk/protolicious/commit/f7b660d 
  There was a typo: you've initialized the `hash` variable, but `current` variable was used in the code above
- I've removed the trailing `.call(null)` block because you're calling it on result of top-level function, that is not returning the function in fact. So the browser displays error message.
- Yeah, the cookie code was optimized for size. I haven't tested it's performance, so i don't know which version will be faster. I think you know better than me which one version is good.
- Element#centerInParent - fixed in c60f39b
